infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/AddNodeWorkflow.java

```diff
@@ -0,0 +1,198 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import com.google.common.collect.ContiguousSet;
+import com.google.common.collect.DiscreteDomain;
+import com.google.common.collect.Range;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.protocols.wireprotocol.RangeWriteMsg;
+import org.corfudb.protocols.wireprotocol.orchestrator.AddNodeRequest;
+import org.corfudb.protocols.wireprotocol.orchestrator.Request;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.LogUnitClient;
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.corfudb.runtime.exceptions.OutrankedException;
+import org.corfudb.runtime.exceptions.QuorumUnreachableException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.CFUtils;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import static org.corfudb.format.Types.OrchestratorRequestType.ADD_NODE;
+
+/**
+ * A definition of a workflow that adds a new node to the cluster.
+ *
+ * @author Maithem
+ */
+@NotThreadSafe
+@Slf4j
+public class AddNodeWorkflow implements Workflow {
+
+    final AddNodeRequest request;
+
+    private Layout newLayout;
+
+    public AddNodeWorkflow(Request request) {
+        this.request = (AddNodeRequest) request;
+    }
+
+    @Override
+    public String getName() {
+        return ADD_NODE.toString();
+    }
+
+    @Override
+    public List<Action> getActions() {
+        return Arrays.asList(new BootstrapNode(),
+                new AddNodeToLayout(),
+                new StateTransfer(),
+                new MergeSegments());
+    }
+
+    class BootstrapNode implements Action {
+        @Override
+        public String getName() {
+            return "BootstrapNode";
+        }
+
+        @Override
+        public void execute(@Nonnull CorfuRuntime runtime) throws Exception {
+            for (; ; ) {
+                try {
+                    runtime.getLayoutManagementView().bootstrapNewNode(request.getEndpoint());
+                    return;
+                } catch (NetworkException e) {
+                    log.error("Network exception while bootstraping new node", e);
+                    Thread.sleep(30 * 1000);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2184)